src/usd3/USD3.sol

```diff
@@ -189,59 +217,62 @@ contract USD3 is BaseHooksUpgradeable {
     function _deployFunds(uint256 _amount) internal override {
         if (_amount == 0) return;
 
-        uint256 maxOnCreditRatio = maxOnCredit();
+        // Wrap USDC to waUSDC
+        _amount = WAUSDC.deposit(_amount, address(this));
 
+        uint256 maxOnCreditRatio = maxOnCredit();
         if (maxOnCreditRatio == 0) {
-            // Don't deploy anything when set to 0%
+            // Don't deploy anything when set to 0%, keep all waUSDC local
             return;
         }
 
-        if (maxOnCreditRatio == 10_000) {
-            // Deploy everything when set to 100%
-            morphoCredit.supply(_marketParams, _amount, 0, address(this), "");
-            return;
-        }
+        // Calculate total waUSDC (deployed + local)
+        uint256 deployedWaUSDC = suppliedWaUSDC();
+        uint256 localWaUSDC = balanceOfWaUSDC();
+        uint256 totalWaUSDC = deployedWaUSDC + localWaUSDC;
 
-        uint256 totalValue = TokenizedStrategy.totalAssets();
-        uint256 maxDeployable = (totalValue * maxOnCreditRatio) / 10_000;
-        uint256 currentlyDeployed = morphoCredit.expectedSupplyAssets(_marketParams, address(this));
+        // Calculate max that should be deployed to MorphoCredit
+        uint256 maxDeployableWaUSDC = (totalWaUSDC * maxOnCreditRatio) / 10_000;
 
-        if (currentlyDeployed >= maxDeployable) {
-            // Already at max deployment
+        if (maxDeployableWaUSDC <= deployedWaUSDC) {
+            // Already at or above max, keep all new waUSDC local
             return;
         }
 
-        uint256 deployableAmount = maxDeployable - currentlyDeployed;
-        uint256 toDeploy = Math.min(_amount, deployableAmount);
+        // Deploy only the amount needed to reach max
+        uint256 waUSDCToSupply = Math.min(localWaUSDC, maxDeployableWaUSDC - deployedWaUSDC);
 
-        if (toDeploy > 0) {
-            morphoCredit.supply(_marketParams, toDeploy, 0, address(this), "");
-        }
+        _supplyToMorpho(waUSDCToSupply);
     }
 
     /// @dev Withdraw funds from MorphoCredit market
-    /// @param amount Amount of asset to free up
-    function _freeFunds(uint256 amount) internal override {
-        if (amount == 0) return;
+    /// @param _amount Amount of asset to free up
+    function _freeFunds(uint256 _amount) internal override {
+        if (_amount == 0) {
+            return;
+        }
 
-        morphoCredit.accrueInterest(_marketParams);
```

Comment:
why are  we removing this accrual

Reply:
we didn't remove it, it moved to the `_withdrawFromMorpho` function